aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -856,7 +856,7 @@ std::tuple<Tensor, Tensor, Tensor> _lu_with_info_cuda(const Tensor& self, bool p
   squareCheckInputs(self);
   auto req_size = self.sizes().vec();
   req_size.pop_back();
-  Tensor pivots_tensor = at::zeros(req_size, self.options().dtype(at::kInt));
+  Tensor pivots_tensor = at::arange(1, req_size[1]+1, self.options().dtype(at::kInt)).unsqueeze(0).expand(req_size);
```

Comment:
req_size includes batch dimensions, and hence this will fail on tensors with greater than 4 dimensions.

This snippet should work fine:
```
Tensor pivots_tensor = at::arange(1, self.size(-1) + 1, self.options().dtype(at::kInt)).expand_as(req_size);
```

Also, a test would be nice.

Reply:
I've updated to that, and have added tests. Still untested on my machine because other build issues.

Reply:
You can set USE_MKLDNN=0 and try building if #21700 is bothering you :)

Reply:
I quickly tried that yesterday and cleaned+rebuilt but still hit that issue

Reply:
Also why would the LU factorizations/solves be called on tensors with 4 or more dimensions? In the batched case, `A` is `[n_batch, n, m]` and `b` is ~`[n_batch, m, n_rhs]`, right?

Reply:
(Maybe yesterday I didn't properly clean my PyTorch build when trying to disable that)

Reply:
3 dimensions would incur a reshaping cost if the input tensor is > 3 dimensions.

Reply:
@bamos I'm sorry. It should've been `expand` instead of `expand_as`, and that has caused the failures.

Reply:
Ah sorry for trying to send this in without a working build, just updated that

Reply:
Thank you, I'll wait for CI to finish.